src/server/game/AI/SmartScripts/SmartScript.cpp

```diff
@@ -1872,6 +1894,20 @@ void SmartScript::ProcessAction(SmartScriptHolder& e, Unit* unit, uint32 var0, u
 
         WorldObject* target = NULL;
 
+        if (e.GetTargetType() == SMART_TARGET_RANDOM_POINT)
+        {
+            if (me)
+            {
+                float range = (float)e.target.randomPoint.range;
+                Position randomPoint;
+                Position srcPos = { e.target.x, e.target.y, e.target.z, e.target.o };
+                me->GetRandomPoint(srcPos, range, randomPoint);
+                me->GetMotionMaster()->MovePoint(e.action.MoveToPos.pointId, randomPoint.m_positionX, randomPoint.m_positionY, randomPoint.m_positionZ, true, true, e.action.MoveToPos.controlled ? MOTION_SLOT_CONTROLLED : MOTION_SLOT_ACTIVE);
```

Comment:
this line is quite long, what about formatting it a bit differently for better readability?

```
me->GetMotionMaster()->MovePoint(
        e.action.MoveToPos.pointId,
        randomPoint.m_positionX,
        andomPoint.m_positionY,
        randomPoint.m_positionZ,
        true,
        true,
        e.action.MoveToPos.controlled ? MOTION_SLOT_CONTROLLED : MOTION_SLOT_ACTIVE
);
```